crates/sui-core/src/authority_server.rs

```diff
@@ -451,6 +470,117 @@ impl ValidatorService {
         Ok((tonic::Response::new(info), Weight::zero()))
     }
 
+    async fn handle_transaction_v2(
+        &self,
+        request: tonic::Request<HandleTransactionRequestV2>,
+        // TODO: Do we want to make a new response type for this?
+    ) -> WrappedServiceResponse<HandleCertificateResponseV3> {
```

Comment:
It seems more refactoring will be needed to support soft bundle. Most of the logic here should be shared between handle_transaction_v2 and handle_soft_bundle_transactions_v2 handlers. Or we can take the opportunity to support batched transaction submission with a single handler, which should cover the soft bundle use case as well.

Reply:
Yes, I was not originally planning to support soft bundle for the first prototype. do you think that's worth adding to this PR?

Reply:
As discussed, we can definitely can back and refactor later.